### PR TITLE
fixed grouped status

### PR DESCRIPTION
**Pull request by xh3b4sd**
I noticed that grouping the status of groups does not work properly. This PR fixes the problem.

### CODE REVIEW

File: controller/status.go [hunk 1]
```diff
@@ -16,6 +16,8 @@ func (usl UnitStatusList) Group() ([]fleet.UnitStatus, error) {
 	matchers := map[string]struct{}{}
 	newList := []fleet.UnitStatus{}
 
+	hashesEqual := allHashesEqual(usl)
```
Comment (xh3b4sd) [inline controller/status.go:19]: This is the trick. When we added unit hashes we compared the unit hashes of the unit files that were already grouped. Obviously it makes no sense to compare unit hashes with the units within the same group. We are interested in comparing the unit hashes across the slices.

**Comment by xh3b4sd** [comment]
RFR @zeisss @Nesurion

File: controller/status.go [hunk 2]
```diff
@@ -63,8 +67,27 @@ func groupUnitStatus(usl []fleet.UnitStatus, groupMember fleet.UnitStatus) ([]fl
...
 }
 
+// allHashesEqual is supposed to receive a list of unit statuses that is not
+// grouped. This is necessary to compare unit hashes across groups.
+func allHashesEqual(usl []fleet.UnitStatus) bool {
+	for _, us1 := range usl {
+		for _, us2 := range usl {
+			for _, m1 := range us1.Machine {
+				for _, m2 := range us2.Machine {
+					if common.UnitBase(us1.Name) == common.UnitBase(us2.Name) && m1.UnitHash != m2.UnitHash {
```
Comment (zeisss) [inline controller/status.go]: can we move the `.Name` comparision two loops earlier?

**Comment by zeisss** [comment]
LGTM, although I would change the check to do less looping.

File: controller/status.go (hunk 2 above)
Comment (xh3b4sd) [inline controller/status.go]: Done

**Pull request merged**